ttools/callbacks.py

```diff
@@ -506,6 +515,22 @@ def batch_end(self, batch_data, train_step_data):
         self.checkpointer.save(filename, extras={"epoch": self.epoch})
         self.__purge_old_files()
 
+    def validation_end(self, val_data):
+        """Save a best model checkpoint if value for best_val_key is lowest so far."""
+
+        super(CheckpointingCallback, self).validation_end(val_data)
+
+        if self.best_val_key is None:
+            return
+        if val_data[self.best_val_key] > self.best_val_value:
```

Comment:
we should probably document that the "best" is implicitly a min value (e.g. if someone uses accuracy instead of loss)